### PR TITLE
add ability to use JSON data to authenticate a user

### PR DESCRIPTION
**Pull request by danielkay**

### CODE REVIEW

**Comment by gfreeau** [comment]
Thank you I am reviewing it

**Comment by danielkay** [comment]
cool no probs- want me to flatten the commits?

**Comment by noliva** [comment]
any news about this PR

**Comment by gfreeau** [comment]
@danielkay Yes, it would be great if you could flatten the commits.

**Comment by danielkay** [comment]
Done :)

**Comment by gfreeau** [comment]
Thanks, I am going to get this into the next release.

Question: What is the reason for catching the ContextErrorException?

**Comment by danielkay** [comment]
Ooops, my bad- was using that during a bit of debugging our symfony app and evidently forgot to remove it. I'll update the PR and remove it asap ;)

**Comment by danielkay** [comment]
Cool, try/catch removed and PR updated :)

**Comment by havvg** [comment]
Ping @gfreeau anything left to do? Looking forward to get this merged :)

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/.idea
```
Comment (havvg) [inline .gitignore]: This should be added to the global or user specific `.gitignore`, not to the project's one.

File: Security/Firewall/GetJWTListener.php [hunk 2]
```diff
@@ -65,17 +66,26 @@ public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
 
-        if ($this->options['post_only'] && !$request->isMethod('POST')) {
-            $event->setResponse(new JsonResponse('invalid method', 405));
-            return;
+        if ($request->headers->get('content_type') == 'application/json') {
```
Comment (havvg) [inline Security/Firewall/GetJWTListener.php]: This check overwrites the `post_only` option, as it allows other verbs, too.

**Comment by gfreeau** [comment]
I will have time to merge everything next week after some more testing on my part.

**Comment by gfreeau** [comment]
@danielkay As @havvg said, some of the code here overwrites the post_only option. This would cause problems  when merging #9 as well. @teohhanhui 

Are you both able to collaborate on making sure both PRs work together? I do not have time right now.

**Comment by danielkay** [comment]
@gfreeau: Apologies - have been away for a while. I'll take a look and see if I can get it sorted :)

**Pull request closed without merging**